pandas/tests/frame/test_alter_axes.py

```diff
@@ -193,19 +190,21 @@ def test_set_index_pass_arrays_duplicate(self, frame_of_index_cols, drop,
         df.index.name = index_name
 
         keys = [box1(df['A']), box2(df['A'])]
-        result = df.set_index(keys, drop=drop, append=append)
 
-        # need to adapt first drop for case that both keys are 'A' --
-        # cannot drop the same column twice;
-        # use "is" because == would give ambiguous Boolean error for containers
-        first_drop = False if (keys[0] is 'A' and keys[1] is 'A') else drop
+        # == gives ambiguous Boolean for Series
```

Comment:
Can you explain the diff here in a little more detail? Was this failing before?

Reply:
This is the repeated drop bug that we discussed in the other issue.

```python
In [18]: df = pd.DataFrame({"A": [1, 2, 3, 4]})

In [19]: df.set_index(['A', 'A'])
```

Reply:
This was already in master before the reverted PR, so let's leave it at this